script.js

```diff
@@ -1,4 +1,390 @@
+/* Note: This implementation incldues the following features:
+1. Regular beat that game (i.e. largest num wins) 
+2. Auto-Generate Combined Number
+3. Variable Number of Dice
+4. Variable Number of Players 
+We've removed Lowest Combined Number to reduce bloat, making the code more legible. However, please feel free to try to implement all the features together on your own
+*/
+/*====================================
+==============GLOBAL VARIABLES========
+======================================*/
+// Game modes
+var GAME_MODE_CHOOSE_NUM_PLAYERS = 'GAME_MODE_CHOOSE_NUM_PLAYERS';
+var GAME_MODE_CHOOSE_NUM_DICE = 'GAME_MODE_CHOOSE_NUM_DICE';
+var GAME_MODE_DICE_ROLL = 'GAME_MODE_DICE_ROLL';
+var GAME_MODE_CHOOSE_DICE_ORDER_AUTOMATICALLY =
+  'GAME_MODE_CHOOSE_DICE_ORDER_AUTOMATICALLY';
+
+// Track num of dice user has chosen to play with
+var numDiceChosen = 0;
+
+// Track num of players
+var numPlayers = 0;
+
+// Initialise the game to start with the dice roll game mode
+var gameMode = GAME_MODE_CHOOSE_NUM_PLAYERS;
+
+// Keep track of the current player's number, either 1 or 2.
+// The game starts with Player 1.
+var currPlayer = 0;
+
+// Set an array that will contains details about each player (i.e. an array of objects)
+var playerProfiles = [];
+
+// Track the current round's winner
+var roundWinner = null;
+
+/*====================================
+==============HELPER FUNCTIONS========
+======================================*/
```

Comment:
See comment in other PR on sections. I don't think these sections are necessary.